src/store/dataset.ts

```diff
@@ -170,16 +172,18 @@ class DatasetStore {
   }
 
   removeCondition({ subGroup, itemName }: IRemoveConditionItem) {
-    const cloneConditions = cloneDeep(this.conditions)
+    const cloneConditions: Condition[] = cloneDeep(this.conditions)
 
     const subGroupIndex = cloneConditions.findIndex(
       item => item[1] === subGroup,
     )
 
-    const conditionKind = cloneConditions.find(item => item[1] === subGroup)[0]
+    const conditionKind = cloneConditions.find(
+      item => item![1] === subGroup,
+    )![0]
 
     if (conditionKind === FilterKindEnum.Enum) {
-      const filteredItems = cloneConditions[subGroupIndex][3].filter(
+      const filteredItems: string[] = cloneConditions[subGroupIndex][3]!.filter(
```

Comment:
remove type assertion and replace ! with ?

Reply:
made edits through rebase so you wont see updates lol